src/helperFunctions/compare_sets.py

```diff
@@ -10,14 +19,37 @@ def remove_duplicates_from_unhashable(unhashable_list: List[T]) -> List[T]:
 
 
 def remove_duplicates_from_list(list_):
+    '''
+    Remove duplicates from a list (by converting it to a set and back).
+
+    .. caution::
+        Only works with lists of hashable objects!
+
+    :param list_: the input list (possibly) containing duplicates
+    :return: a new list only containing unique elements
+    '''
     return list(set(list_))
 
 
 def substring_is_in_list(string: str, substring_list: List[str]) -> bool:
+    '''
+    Check if any element in a list of strings is a substring of the provided string.
+
+    :param string: the string that is checked if it contains any of the substrings
+    :param substring_list: a list of possible substrings
+    :return: true if a substring is found and false otherwise
+    '''
     return any(substring in string for substring in substring_list)
 
 
-def iter_element_and_rest(iterable: Iterable[T]) -> Tuple[T, Iterable[T]]:
+def iter_element_and_rest(iterable: Iterable[_T]) -> Iterator[Tuple[_T, Iterable[_T]]]:
+    '''
+    Iterate over each element of an iterable object (e.g. a list) and also get all other (remaining) elements
+    from the object.
+
+    :param iterable: the object that will be iterated over
+    :return: a generator with tuples containing the item and the rest (all other elements)
+    '''
     for element in iterable:
         rest = [e for e in iterable if e != element]
         yield element, rest
```

Comment:
This should be collapsed to one line (while you're at it)

Reply:
reduced it to two lines but I would prefer to keep it a generator, because the lists may be big and it's more efficient this way

Reply:
That's what I meant. One (yield) line inside the for-loop.